File: src/main/webapp/app/core/admin/passkey-management/admin-passkey.dto.ts
export interface AdminPasskeyDTO {
    /**
     * The credential ID of the passkey, encoded in Base64url format.
     */
    credentialId: string;
    label: string;

    /**
     * ISO 8601 date string
     */
    created: string;

    /**
     * ISO 8601 date string
     */
    lastUsed: string;

    /**
     * Super admin approval is required for privileged features (e.g. administrator features)
     */
    isSuperAdminApproved: boolean;

    /**
     * The ID of the user who owns this passkey
     */
    userId: number;

    /**
     * The login/username of the user who owns this passkey
     */
    userLogin: string;

    /**
     * The full name of the user who owns this passkey
     */
    userName: string;
}

File: src/main/java/de/tum/cit/aet/artemis/core/dto/AdminPasskeyDTO.java
package de.tum.cit.aet.artemis.core.dto;

import java.time.Instant;

import jakarta.validation.constraints.NotNull;

import com.fasterxml.jackson.annotation.JsonInclude;

import de.tum.cit.aet.artemis.core.config.validator.Base64Url;

/**
 * A DTO representing a passkey credential with user information for administrative purposes.
 * This extends the basic PasskeyDTO with additional user details.
 */
@JsonInclude(JsonInclude.Include.NON_EMPTY)
public record AdminPasskeyDTO(@NotNull @Base64Url String credentialId, String label, Instant created, Instant lastUsed, boolean isSuperAdminApproved, Long userId, String userLogin,
        String userName) {
}
